src/UsblParser.cpp

```diff
@@ -153,7 +153,7 @@ ReceiveIM UsblParser::parseReceivedIM(string const &buffer)
         im.deliveryReport = true;
     else
         im.deliveryReport = false;
-    im.duration = base::Time::fromMicroseconds(stod(splitted[5],&sz));
+    im.duration = base::Time::fromMicroseconds(stoi(splitted[5],&sz));
```

Comment:
The index argument (the second one) to `stoi()` is optional, if you are not using it you may just not pass it..